source/Mlos.Model.Services.Client/OptimizerServiceEncoderDecoder.cs

```diff
@@ -0,0 +1,419 @@
+            }
+
+            instance.Name = hypergrid.Name;
+
+            return instance;
+        }
+
+        public OptimizerService.OptimizationProblem EncodeOptimizationProblem(OptimizationProblem problem)
+        {
+            OptimizerService.OptimizationProblem instance = new OptimizerService.OptimizationProblem();
+            instance.ParameterSpace = EncodeHypergrid(problem.ParameterSpace);
+            instance.ObjectiveSpace = EncodeHypergrid(problem.ObjectiveSpace);
+            foreach (var objective in problem.Objectives)
+            {
+                instance.Objectives.Add(EncodeObjective(objective));
+            }
+
+            if (problem.ContextSpace == null)
+            {
+                instance.EmptyContext = new OptimizerService.Empty();
+            }
+            else
+            {
+                instance.ContextSpace = EncodeHypergrid(problem.ContextSpace);
+            }
+
+            return instance;
+        }
+
+        public OptimizerService.Objective EncodeObjective(OptimizationObjective objective)
+        {
+            return new OptimizerService.Objective
+            {
+                Minimize = objective.Minimize,
+                Name = objective.Name,
+            };
+        }
+
+        public OptimizerService.GuestSubgrid EncodeSubgridJoin(SubgridJoin subgridJoin)
+        {
+            return new OptimizerService.GuestSubgrid
+            {
+                ExternalPivotDimension = EncodeDimension(subgridJoin.OnExternalJoin),
+                Subgrid = EncodeHypergrid(subgridJoin.Subgrid),
+            };
+        }
+
+        private OptimizerService.DimensionType DimensionToGrpcType(DimensionTypeName typeName)
+        {
+            switch (typeName)
+            {
+                case DimensionTypeName.CategoricalDimension:
+                    return OptimizerService.DimensionType.Categorical;
+                case DimensionTypeName.ContinuousDimension:
+                    return OptimizerService.DimensionType.Continuous;
+                case DimensionTypeName.DiscreteDimension:
+                    return OptimizerService.DimensionType.Discrete;
+                case DimensionTypeName.OrdinalDimension:
+                    return OptimizerService.DimensionType.Ordinal;
+                default:
+                    System.Console.Error.WriteLine("Invalid type provided to OptimizerServiceEncoderDecoder.Encoder.DimensionToGrpcType(..). Type: " + typeName);
+                    return OptimizerService.DimensionType.Continuous;
+            }
+        }
+
+        public OptimizerService.Dimension EncodeDimension(IDimension dimension)
+        {
+            if (dimension is EmptyDimension dimension1)
+            {
+                return new OptimizerService.Dimension
+                {
+                    EmptyDimension = EncodeEmptyDimension(dimension1),
+                };
+            }
+            else if (dimension is ContinuousDimension dimension2)

```

Comment:
if (dimension is ContinuousDimension continuousDimension )

Reply:
[zaporter/grpc 630599a6] Improved readability of if (var is Type varname) and removed old comment